lib/rust/data-structures/src/diet.rs

```diff
@@ -143,12 +143,9 @@ pub struct $name {
 
 impl $name {
 
-    /// Create an empty data array. This function is safe because the intervals are build out of
-    /// [`uint`]s, which can be initialized from raw memory. Follow the link to learn more:
-    /// https://doc.rust-lang.org/std/mem/union.MaybeUninit.html.
-    #[allow(unsafe_code)]
+    /// Create an empty data array.
     pub (crate) fn empty_data_array() -> DataArray {
-        unsafe { MaybeUninit::uninit().assume_init() }
+        [default();$num]
     }
```

Comment:
This changes the implementation. If we want to use array literals, we do not need the function. Are you sure it has the same performancei mplications as the previous version? Who was the author of the prev version - have you been chatting with them about this change?

Reply:
You are right, the docs are not correct. However, I'm not sure whether this data structure ever tries to access elements that were not initialized by some logic in this structure. Initializing all elements upfront can cause performance drop, and this is a low-level struct. As we are not using it anywhere in our codebase (we are just referencing it from the docs), I'd propose to chose the first solution described below. I see 2 possible solutions here:

1. Leaving it as it was + fixme about wrong docs + potential UB to be checked one day.
2. Fixing it properly - this includes analysis if these elements really need to be initialized, benchmarking vs the old version, and fixing docs.

Reply:
> Note that the UB is not caused by *accessing* but merely by initialising. The correct solution would have to propagate the `MaybeUninit` until after initialisation. I don't think this should be part of this PR, instead we should file a new task to investigate this. We are hotfixing UB here, aplying a new performance improvement is somewhat less urgent.